code/datums/components/mechComp_signals.dm

```diff
@@ -308,7 +308,7 @@ TYPEINFO(/datum/component/mechanics_holder)
 		boutput(user, SPAN_ALERT("Components need to be within a range of 14 meters to connect."))
 		return
 
-	var/typesel = input(user, "Use [parent] as:", "Connection Type") in list("Trigger", "Receiver", "*CANCEL*")
+	var/typesel = tgui_input_list(user, "Use [parent] as:", "Connection Type", list("Trigger", "Receiver", "*CANCEL*"))
```

Comment:
i don't know if it makes sense to use lists and not alerts for 3/2 options

Reply:
Sure